allure-generator/src/main/javascript/components/node-sorter/NodeSorterView.js

```diff
@@ -0,0 +1,80 @@
+import './styles.css';
+import {on, className} from '../../decorators';
+import settings from '../../util/settings';
+import template from './NodeSorterView.hbs';
+import {View} from 'backbone.marionette';
+import {values} from '../../util/statuses';
+
+
+@className('sorter')
+class NodeSorterView extends View {
+    template = template;
+
+    sorters = [
+        {
+            key: 'name',
+            sorter: (a, b, c) => {return a.name.toLowerCase() < b.name.toLowerCase() ? -c : c;}
+        },
+        {
+            key: 'duration',
+            sorter: (a, b, c) => {return a.time.duration < b.time.duration ? -c : c;}
+        },
+        {
+            key: 'status',
+            sorter: (a, b, c) => {
+                if ('status' in a && 'status' in b){
+                    return values.indexOf(a.status) > values.indexOf(b.status) ? -c : c;
+                } else if ('statistic' in a && 'statistic' in b){
+                    return values.reduce((all, current) => {
+                        if ((a.statistic[current] !== b.statistic[current]) && all === 0) {
+                            return b.statistic[current] > a.statistic[current];
+                        } else {
+                            return all;
+                        }
+                    }, 0) ? -c: c;
+                } else {
+                    return 1;
+                }
+            }
+        },
+    ];
+
+    initialize({sorterSettingsKey}) {
+        this.sorterSettingsKey = sorterSettingsKey;
+    }
+
+    getSorter(){
+        const sortSettings = settings.getTreeSorting(this.sorterSettingsKey);
+        const sorter = this.sorters[sortSettings.sorter].sorter;
+        const direction =  sortSettings.ascending ? 1 : -1;
+        return (a, b) => sorter(a, b, direction);
+    }
+
+    @on('click .sorter__item')
+    onChangeSorting(e){
+        const el = this.$(e.currentTarget);
+        settings.save(this.sorterSettingsKey, {
+            sorter: el.data('index'),
+            ascending: !el.data('asc')
```

Comment:
`!el.data('asc')` repeated a lot there.

Will be better to make it constant at the top:

```js
const asc = !el.data('asc');
```